### PR TITLE
feat: formatter utils

### PR DESCRIPTION
**Pull request by enitrat**
Adds a set of utils to Display/Debug types not supported in the core library.
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,8 +11,6 @@ jobs:
         uses: actions/checkout@v3
       - name: Step 2 - Getting scarb
         uses: software-mansion/setup-scarb@v1
```
Comment (gaetbout) [inline .github/workflows/test.yml:13]: Could prob update this too to use latest `uses: software-mansion/setup-scarb@v1.3.2`

File: Scarb.toml [hunk 2]
```diff
@@ -11,14 +11,15 @@ members = [
...
+  "src/utils",
 ]
 name = "alexandria"
 version = "0.1.0"
 description = "Community maintained Cairo and Starknet libraries"
 homepage = "https://github.com/keep-starknet-strange/alexandria/"
 
 [workspace.dependencies]
-starknet = ">=2.4.0-rc2"
+starknet = ">=2.4.0-rc6"
```
Comment (gaetbout) [inline Scarb.toml:22]: 2.4.0?

**Comment by enitrat** [comment]
done

File: .github/workflows/test.yml (hunk 1 above)
Comment (gaetbout) [reply .github/workflows/test.yml:13]: Referring to the scarb action version here 😄

Comment (enitrat) [reply .github/workflows/test.yml:13]: ah, ok

File: src/utils/Scarb.toml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "alexandria_utils"
+version = "0.3.0"
```
[the remark this answers is not in the document]
Comment (enitrat) [reply src/utils/Scarb.toml:3]: Yes

[the remark this answers is not in the document]
Comment (enitrat) [reply src/utils/Scarb.toml:3]: I don't think it fits anywhere else

[the remark this answers is not in the document]
Comment (enitrat) [reply src/utils/Scarb.toml:3]: ah this is a mistake sorry

File: src/utils/src/fmt.cairo [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+/// Display
+mod display_felt252_based {
+    use core::fmt::{Display, Formatter, Error};
+    use core::to_byte_array::AppendFormattedToByteArray;
+    impl TDisplay<T, +Into<T, felt252>, +Copy<T>> of Display<T> {
+        fn fmt(self: @T, ref f: Formatter) -> Result<(), Error> {
+            let value: felt252 = (*self).into();
+            Display::<u256>::fmt(@value.into(), ref f)?;
+            Result::Ok(())
```
Comment (LucasLvy) [inline src/utils/src/fmt.cairo:13]: directly return the result ?

File: src/utils/src/fmt.cairo [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                    }
+                    if write!(f, ", ").is_err() {
+                        break Result::Err(Error {});
+                    };
+                },
+                Option::None => { break Result::Ok(()); }
+            };
+        }?;
+        write!(f, "]")?;
+        Result::Ok(())
```
Comment (LucasLvy) [inline src/utils/src/fmt.cairo:67]: return directly the result ?

File: src/utils/src/fmt.cairo (hunk 4 above)
Comment (enitrat) [reply src/utils/src/fmt.cairo]: Done

File: src/utils/src/fmt.cairo (hunk 5 above)
Comment (enitrat) [reply src/utils/src/fmt.cairo]: Done

**Review by LucasLvy — APPROVED** [review]

**Pull request merged**